### PR TITLE
fix(fe/jig/edit): Prevent cover module being edited before having a cover dragged in

### LINKED ISSUE #1961: Cover module should not open before the user dragged the cover module in
When clicking on the cover module, the module opens. Please make sure that this only happens after the user dragged the module in, otherwise clicks should be ignored.

**Before dragging in the cover module** (should not be clickable in this state)
image

**After dragging in the cover module**
image

### PR DESCRIPTION
**Pull request by johnnynotsolucky**
Resolves #1961

Other changes included:

- Fixes some syntax noted in #1967;
- Refactors MenuDom so that children are constructed by that component instead;
- Add a render function for rendering the modules "add" button.

### CODE REVIEW

File: frontend/apps/crates/entry/jig/edit/src/edit/sidebar/module/dom.rs [hunk 1]
```diff
@@ -62,21 +62,35 @@ impl ModuleDom {
...
+                // TODO:
+                // .event(|_evt:events::MouseDown| {
+                //     actions::mouse_down(state.clone(), evt.x(), evt.y());
+                // })
                 .event_with_options(
                     &EventOptions::bubbles(),
                     clone!(state => move |_evt:events::Click| {
-                        match &*state.module {
+                        let can_edit = match &*state.module {
                             Some(_) => {
```
Comment (johnnynotsolucky) [inline frontend/apps/crates/entry/jig/edit/src/edit/sidebar/module/dom.rs:73]: This resolves #1961

File: frontend/apps/crates/entry/jig/edit/src/edit/sidebar/menu/dom.rs [hunk 2]
```diff
@@ -18,157 +18,199 @@ const STR_PASTE: &'static str = "Paste from another JIG";
...
-        .property("slot", "menu")
-        .child(html!("jig-edit-sidebar-module-menu", {
-            .children(items)
-        }))
-        .event_with_options(&EventOptions::bubbles(), |e: events::Click| {
-            e.stop_propagation();
+pub struct MenuDom;
+
+impl MenuDom {
+    pub fn render(module_state: Rc<ModuleState>) -> Dom {
```
Comment (MendyBerger) [inline frontend/apps/crates/entry/jig/edit/src/edit/sidebar/menu/dom.rs]: @johnnynotsolucky why the move to a struct with a method?

Comment (johnnynotsolucky) [reply frontend/apps/crates/entry/jig/edit/src/edit/sidebar/menu/dom.rs]: @MendyBerger only to get it more or less similar to how other components are built, for example [EditPage](https://github.com/johnnynotsolucky/ji-cloud/blob/sandbox/frontend/apps/crates/entry/jig/edit/src/edit/dom.rs#L17-L20) and [ModuleDom](https://github.com/johnnynotsolucky/ji-cloud/blob/sandbox/frontend/apps/crates/entry/jig/edit/src/edit/sidebar/module/dom.rs#L15-L18). 

Which is preferred for these?

If I should revert, it's minimal work to revert it.

Comment (johnnynotsolucky) [reply frontend/apps/crates/entry/jig/edit/src/edit/sidebar/menu/dom.rs]: Also, these refactors should've been in a separate PR to keep PRs simpler. I'll do that next time.

Comment (MendyBerger) [reply frontend/apps/crates/entry/jig/edit/src/edit/sidebar/menu/dom.rs]: The way components are structured has changed over time, having a Dom struct that has a method was actually before the bare function, but we have now moved on from the bare function as well.

Maybe I can give you a basic rundown of this in the daily meeting.

Comment (johnnynotsolucky) [reply frontend/apps/crates/entry/jig/edit/src/edit/sidebar/menu/dom.rs]: I reverted most of the refactor of MenuDom, I'll make a separate PR later with more up-to-date changes.

**Pull request merged**